iceoryx_posh/include/iceoryx_posh/runtime/posh_runtime.hpp

```diff
@@ -67,16 +67,17 @@ class PoshRuntime
 
     /// @brief find all services that match the provided service description
     /// @param[in] serviceDescription service to search for
-    /// @param[out] instanceContainer container that is filled with all matching instances
-    /// @return cxx::expected<Error> Error, if any, encountered during the operation
+    /// @return cxx::expected<InstanceContainer,Error>
+    /// InstanceContainer: on success, container that is filled with all matching instances
+    /// Error: if any, encountered during the operation
     /// Error::kPOSH__SERVICE_DISCOVERY_INSTANCE_CONTAINER_OVERFLOW : Number of instances can't fit in instanceContainer
     /// Error::kMQ_INTERFACE__REG_UNABLE_TO_WRITE_TO_ROUDI_MQ : Find Service Request could not be sent to RouDi
-    cxx::expected<Error> findService(const capro::ServiceDescription& serviceDescription,
-                                     InstanceContainer& instanceContainer) noexcept;
+    cxx::expected<InstanceContainer, Error> findService(const capro::ServiceDescription& serviceDescription) noexcept;
 
     /// @brief offer the provided service, sends the offer from application to RouDi daemon
     /// @param[in] serviceDescription service to offer
-    void offerService(const capro::ServiceDescription& serviceDescription) noexcept;
+    /// @return bool, if service is offered returns true else false
```

Comment:
Could you please write when this method will return false. E.g. when you are providing an invalid service description.